File: vernam.cs
public string vernam_e(string toE, string pw){
	string enc="";
	for (int k = 0, j = 0; k < toE.Length; ++k) {
		enc += (char)(toE [k] + pw [j]);
		j = (j + 1) % pw.Length;
	}
	return enc;
}
public string vernam_d(string toD, string pw){
	string enc="";
	for (int k = 0, j = 0; k < toD.Length; ++k) {
		enc += (char)(toD [k] - pw [j]);
		j = (j + 1) % pw.Length;
	}
	return enc;
}

File: vernam.ts
export function encode(toE:string, pw:string):string{
  let result:string = '';
  for (let index = 0, vi=0; index < toE.length; ++index) {
      result += String.fromCharCode( toE.charCodeAt(index) + pw.charAt(vi) );
      vi = (vi + 1)%vi.length;
  }
  return result;
}

export function decode(toD:string, pw:string):string{
  let result:string = '';
  for (let index = 0, vi=0; index < toD.length; ++index) {
      result += String.fromCharCode( toD.charCodeAt(index) - pw.charAt(vi) );
      vi = (vi + 1)%vi.length;
  }
  return result;
}
